cnnimg.py

```diff
@@ -65,111 +62,107 @@ def imageprocessing(self, inputlocation, outputlocation, tempA, tempB, initialco
         u = np.array(gray)
         u = u[:,:,0]
         z0 = (u)*initialcondition
-        self.Bu = sig.convolve2d(u, tempB, 'same')
-        self.tempA = tempA
-        self.Ib = Ib
-        z0 = z0.reshape(z0.size)
-        z = self.cnn(sint.odeint(self.f, z0, t))
+        Bu = sig.convolve2d(u, tempB, 'same')
+        z0 = z0.flatten()
+        z = self.cnn(sint.odeint(self.f, z0, t, args=(Ib, Bu, tempA)))
         l = z[z.shape[0]-1,:].reshape((self.n, self.m))
         l = l/(255.0)
-        for i in range(l.shape[0]):
-            for j in range(l.shape[1]):
-                l[i][j] = np.uint8(round(l[i][j]*255))
+        l = np.uint8(np.round(l*255))
```

Comment:
I have no way to test this on Raspberry Pi, so that's that.

One thing I'd suggest is checking `l.dtype` after the `for` loop - the way you implemented it, it stays as a float array, so even though you're converting each element to `uint8` they still get written into float arrays and are treated like floats afterwards.

Using `np.uint8` directly on `l` turns it into an integer array, and it works beautifully from then on - at least on Linux.